### PR TITLE
Algae sim

### PR DESCRIPTION
**Pull request by luxiya01**
An attempt for the algae farm simulation.
- Running `roslaunch smarc_stonefish_sims algae.launch` currently launches the environment in `data/algae_env.scn` with 2 ropes
- Run `main.py` in `data/algae/scripts/` to generate new environment files (specify `-o` to not overwrite the current `data/algae_env.scn`)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+## same as for the library above
+# add_dependencies(${PROJECT_NAME}_node ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+
+## Specify libraries to link a library or executable target against
+# target_link_libraries(${PROJECT_NAME}_node
+#   ${catkin_LIBRARIES}
+# )
+
+#############
+## Install ##
```
Comment (nilsbore) [inline CMakeLists.txt]: You need to add install rules for all the files. See e.g. sam_stonefish_sim for examples.

File: data/algae/algae_cube/0.mtl [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+newmtl algae_green
+Ns 225.000000
+Ka 1.000000 1.000000 1.000000
+Kd 0.800000 0.800000 0.800000
+Ks 0.500000 0.500000 0.500000
+Ke 0.000000 0.000000 0.000000
+Ni 1.450000
+d 1.000000
+illum 2
+map_Kd /home/li/Desktop/smarc/catkin_ws/src/smarc_stonefish_sims/data/algae/textures/algae_green.jpg
```
Comment (nilsbore) [inline data/algae/algae_cube/0.mtl]: I don't think stonefish can use .mtl files. These things should probably be within the scenario files instead.

**Review by nilsbore — COMMENTED** [review]
In general it looks very good! I'm gonna try to run it ASAP!

**Comment by nilsbore** [comment]
The major thing is that all of this should be moved to be within a subfolder/package. Basically move everything except `README.md` to something like `algae_world` folder, and rename the package to the same thing in `CMakeLists.txt` and `package.xml`. Note that you can achieve the moving easily with `git mv`.

**Comment by nilsbore** [comment]
I just looked at the sim. It looks amazing!!! Good job @luxiya01 !

File: algae_world/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,218 @@
+cmake_minimum_required(VERSION 3.0.2)
+project(smarc_stonefish_sims_algae_world)
```
Comment (nilsbore) [inline algae_world/CMakeLists.txt]: Probably rename it just algae_world, in line with the folder name as this is the convention. Also, smarc_stonefish_sims_algae_world is a bit of a moutful :).

File: CMakeLists.txt (hunk 1 above)
Comment (luxiya01) [reply CMakeLists.txt]: I added install rules for the three folders, not entirely sure if we need more install rules right now...?

File: algae_world/CMakeLists.txt (hunk 3 above)
Comment (luxiya01) [reply algae_world/CMakeLists.txt]: That's true :rofl:

**Comment by nilsbore** [comment]
@luxiya01 Are we good to merge?

**Comment by luxiya01** [comment]
@nilsbore I think so! If you think the install rules look ok for now :)

**Comment by nilsbore** [comment]
Have you tried running it from an install space (see https://github.com/smarc-project/rosinstall/blob/master/docs/release_package.md#install-rules)?

**Comment by luxiya01** [comment]
> Have you tried running it from an install space (see https://github.com/smarc-project/rosinstall/blob/master/docs/release_package.md#install-rules)?

Yes! Now I have and it seems to be fine :smiley:

**Comment by nilsbore** [comment]
> > Have you tried running it from an install space (see https://github.com/smarc-project/rosinstall/blob/master/docs/release_package.md#install-rules)?
> 
> Yes! Now I have and it seems to be fine smiley

Good, then everything should work!

**Pull request merged**